Source/Scene/Model.js

```diff
@@ -4094,20 +4018,32 @@ define([
                 }
 
                 this._loadResources = new LoadResources();
-                parse(this);
             }
+            parse(this);
         }
 
         var loadResources = this._loadResources;
         var incrementallyLoadTextures = this._incrementallyLoadTextures;
         var justLoaded = false;
 
         if (this._state === ModelState.LOADING) {
-            // Create WebGL resources as buffers/shaders/textures are downloaded
-            createResources(this, frameState);
-
             // Transition from LOADING -> LOADED once resources are downloaded and created.
             // Textures may continue to stream in while in the LOADED state.
+            if (loadResources.pendingBufferLoads === 0 && loadResources.pendingShaderLoads === 0) {
+                if (!this._updatedGltfVersion) {
+                    updateVersion(this.gltf);
```

Comment:
> The reason it is down here is because buffers need to be loaded before we use updateVersion

Ah, why?  If the buffers are really needed, I wonder if this should be two stages: update JSON and update buffers.

> When the model loader code pulls a resource it also adds a reference to the data here. This was the cleanest approach I could find to be able to use pipeline code without modifying the resource loader.

What is the issue with the previous approach and storing things in `_loadResources`?  We're using an implementation detail of `gltf-pipeline` to store data in those `extras` properties.  Are you sure that the glTF objects need to be augmented at all instead of looking things by id in `_loadResources`?

>> Are the extras removed later?

> We can, we do have a stage for that. The only compelling reason I can think of to do that would be if you were freeing the resource data, the gltf object still has references to the buffers, shaders etc, so the data wouldn't be freed. I don't know that we ever free the data and hold onto the gltf though.

The pipeline stages should still clean up after themselves and not leave this side-effect.